scipy/interpolate/tests/test_fitpack2.py

```diff
@@ -146,10 +146,27 @@ def test_nan(self):
         # bail out early if the input data contains nans
         x = np.arange(10, dtype=float)
         y = x**3
+        w = np.ones_like(x)
+        # also test LSQUnivariateSpline [which needs explicit knots]
+        spl = UnivariateSpline(x, y, check_finite=True)
+        t = spl.get_knots()[3:4]  # interior knots w/ default k=3
```

Comment:
`[3:4]` and the comment is somewhat misleading, given that 

```
In [7]: spl = UnivariateSpline(x, y, check_finite=True)

In [8]: spl.get_knots()
Out[8]: array([ 0.,  9.])
```

(So the idea is to have empty interior knots for the call to LSQUnivariateSpline. Apparently, this works.)

Could be nice to clarify the intent, but this can be made in a separate PR.